src/rewriter/rewrite_db_proof_cons.cpp

```diff
@@ -518,11 +519,60 @@ bool RewriteDbProofCons::proveWithRule(RewriteProofStatus id,
   }
   else if (id == RewriteProofStatus::ARITH_POLY_NORM)
   {
-    if (!theory::arith::PolyNorm::isArithPolyNorm(target[0], target[1]))
+    if (target[0].getType().isBoolean())
```

Comment:
Is it worthwhile splitting `RewriteProofStatus::ARITH_POLY_NORM` to another internal enum id as well?

Reply:
No strong opinion here. I tried to be consistent here by following `CONG + EVAL` approach.